### PR TITLE
docs - update to HA overview topics for WAL rep

### PR DESCRIPTION
**Pull request by mkiyama**
Overview of changes:
--replace filerep information with WALrep information in overview topics
--edit overview topics
--update gp_segment_configuration catalog table
--add gp_stat_replication catalog table

NOTE: This PR is only for the overview topics. Other PRs will be opened for the
procedural topics in the Admin Guide.

Link to HTML format of updated topics on the GPDB Docs Review site.
Admin Guide GPDB Overview topic
http://docs-gpdb-review-staging.cfapps.io/review/admin_guide/intro/about_ha.html

Admin Guide HA Topics
http://docs-gpdb-review-staging.cfapps.io/review/admin_guide/highavail/topics/g-overview-of-high-availability-in-greenplum-database.html
http://docs-gpdb-review-staging.cfapps.io/review/admin_guide/highavail/topics/g-overview-of-master-mirroring.html
http://docs-gpdb-review-staging.cfapps.io/review/admin_guide/highavail/topics/g-overview-of-segment-mirroring.html

System Catalogs
http://docs-gpdb-review-staging.cfapps.io/review/ref_guide/system_catalogs/gp_segment_configuration.html
http://docs-gpdb-review-staging.cfapps.io/review/ref_guide/system_catalogs/gp_stat_replication.html

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            <entry colname="col1"><codeph>pid</codeph></entry>
+            <entry colname="col2">integer</entry>
+            <entry colname="col3"/>
+            <entry colname="col4">Process ID of WAL sender backend process. </entry>
+          </row>
+          <row>
+            <entry colname="col1"><codeph>usesysid</codeph></entry>
+            <entry colname="col2">integer</entry>
+            <entry colname="col3"/>
+            <entry colname="col4">User system ID that runs the WAL sender backend process</entry>
```
Comment (litzell) [inline gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml]: add a period after 'backend process'

File: gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-master-mirroring.xml [hunk 2]
```diff
@@ -5,34 +5,47 @@
     <title>Overview of Master Mirroring</title>
     <body>
         <p>You can deploy a backup or mirror of the master instance on a separate host machine or on
```
Comment (litzell) [inline gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-master-mirroring.xml]: "You can deploy a backup or mirror of the master instance on a separate host machine or on the same host machine." 

Although I probably wrote that back when, I don't think it's true. They need to be on separate hosts to have the same port number.  Even if it's somehow true, I'm not sure we should say so.

**Comment by aa77m7a** [comment]
You’re receiving notifications because you’re watching this repository.

File: gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-master-mirroring.xml [hunk 3]
```diff
@@ -5,34 +5,47 @@
...
-                height="165px"/>
+                height="165px" id="image_xrf_qpw_2fb"/>
         </fig>
-        <p>If the primary master fails, the replication process stops, and an administrator can
-            activate the standby master. Upon activation of the standby master, the replicated logs
-            reconstruct the state of the primary master at the time of the last successfully
-            committed transaction. The activated standby then functions as the Greenplum Database master, accepting connections on the port specified when
-            standby master was initialized.</p>
+        <p>Since the master does not house user data, only system catalog tables are synchronized
+            between the primary and standby masters. When these tables are updated, the WAL logs
```
[the remark this answers is not in the document]
Comment (dyozie) [reply gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-master-mirroring.xml]: Agree - It doesn't seem like we use the term so much that an acronym is necessarily needed, although I think it's good to introduce the acronym somewhere to tie-in with postgres terminology.  Better to say "write-ahead log" or even later, "log replication" to save some frustration.

File: gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-segment-mirroring.xml [hunk 4]
```diff
@@ -4,70 +4,77 @@
...
-            segment configurations. Custom mirroring configurations can be described in a
-            configuration file and passed on the command line. </p>
+        <p>If segment mirroring is not enabled, the Greenplum Database system shuts down if a
+            segment instance fails. Administrators manually recover all failed segments before
+            Greenplum Database operations resume.</p>
+        <p>When a segment mirroring is enabled for an existing system, the primary segment instances
+            continue provide service to users while a snapshot of the primary segments are taken.
+            While the snapshots are taken and deployed on the mirror segment instances, changes to
+            the primary segment are also recorded. After the snapshot has been deployed on the
+            mirror segment, the mirror segment is synchronized and kept up to date using Write-Ahead
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-segment-mirroring.xml]: up to date -> up-to-date

File: gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-segment-mirroring.xml [hunk 5]
```diff
@@ -4,70 +4,77 @@
...
+            Logging (WAL)-based streaming replication. Greenplum Database WAL replication uses the
+                <codeph>walsender</codeph> and <codeph>walreceiver</codeph> replication processes.
+            The <codeph>walsender</codeph> process is a primary segment process. The
+                <codeph>walreceiver</codeph> is a mirror segment process. </p>
+        <p>When database changes occur, the WAL logs that capture the changes are copied to the
+            mirror segment to keep it current with the primary segments. During WAL replication,
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-segment-mirroring.xml]: the primary segments -> the corresponding primary segment

or "its primary segment"

File: gpdb-doc/dita/admin_guide/intro/about_ha.xml [hunk 6]
```diff
@@ -19,62 +19,62 @@
...
-          otherprops="pivotal"> and required for Pivotal support</ph>. </p>
-      <p>The secondary (mirror) segment must always reside on a different host than its primary
-        segment to protect against a single host failure. In virtualized environments, the
-        secondary (mirror) must always reside on a different storage system than the master.
-        Mirror segments can be arranged over the remaining hosts in the cluster in configurations
-        designed to maximize availability, or minimize the performance degradation when hosts or
-        multiple primary segments fail. </p>
+      <p>When you deploy your Greenplum Database system, you can configure <i>mirror</i> segment
+        instances. Mirror segments allow database queries to fail over to a backup segment if the
+        primary segment becomes unavailable. The mirror segment is kept up to date by a transaction
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/intro/about_ha.xml]: up-to-date

(global)

File: gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-high-availability-in-greenplum-database.xml [hunk 7]
```diff
@@ -91,35 +90,37 @@
...
       <section id="master_mirroring">
          <title>Master Mirroring</title>
-         <p>There are two masters in a highly available cluster, a <i>primary</i> and a
+         <p>There are two master instances in a highly available cluster, a <i>primary</i> and a
                <i>standby</i>. As with segments, the master and standby should be deployed on
             different hosts so that the cluster can tolerate a single host failure. Clients connect
             to the primary master and queries can be executed only on the primary master. The
-            secondary master is kept up-to-date by replicating the write-ahead log (WAL) from the
-            primary to the secondary. </p>
+            standby master is kept up to date with the primary master using Write-Ahead Logging
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-high-availability-in-greenplum-database.xml]: up to date -> up-to-date

File: gpdb-doc/dita/admin_guide/intro/about_ha.xml (hunk 6 above)
Comment (dyozie) [reply gpdb-doc/dita/admin_guide/intro/about_ha.xml]: This instance also uses "transaction log replication process", which seems like a fine replacement for other spots that use "Write-Ahead-Log (WAL)-replication"

File: gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+            <entry colname="col3">references</entry>
+            <entry colname="col4">description</entry>
+          </row>
+        </thead>
+        <tbody>
+          <row>
+            <entry colname="col1"><codeph>gp_segment_id</codeph></entry>
+            <entry colname="col2">integer</entry>
+            <entry colname="col3"/>
+            <entry colname="col4">The unique identifier of a segment (or master) instance.</entry>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml]: The unique -> Unique

(None of the other descriptions start with articles.)

File: gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+            <entry colname="col2">timestamp </entry>
+            <entry colname="col3"/>
+            <entry colname="col4">Operation start timestamp. </entry>
+          </row>
+          <row>
+            <entry colname="col1"><codeph>state</codeph></entry>
+            <entry colname="col2">text</entry>
+            <entry colname="col3"/>
+            <entry colname="col4">WAL sender state. The value can
+                  be:<p><codeph>startup</codeph></p><p><codeph>backup</codeph> - is this in
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml]: Let's just make sure this question is resolved & removed before the merge.  @Eulerizeit can you answer?

File: gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml [hunk 10]
```diff
@@ -0,0 +1,132 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic
+  PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
+<topic id="topic1" xml:lang="en">
+  <title id="op164582">gp_stat_replication</title>
+  <body>
+    <!--6.0 table -WAL replication for both master and segments-->
+    <p>The <codeph>gp_stat_replication</codeph> view contains metadata of the
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml]: I think it's necessary to use the WAL abbreviation throughout the table as you have, but let's re-define what WAL stands for here in the table description somewhere.

**Comment by dyozie** [comment]
I'd be fine with removing most of this.  As long as the
gp_segment_configuration / gpstate status information is available
somewhere.  I think that info would be valuable to provide in some format,
to understand where things are during segment recovery.

File: gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-master-mirroring.xml [hunk 11]
```diff
@@ -5,34 +5,47 @@
...
+            that capture the changes are copied to the standby master to keep it current with the
```
Comment (jimmyyih) [inline gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-master-mirroring.xml]: "the changes are copied" could be changed to "the changes are streamed"

File: gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-segment-mirroring.xml [hunk 12]
```diff
@@ -4,70 +4,77 @@
...
+            mirror segment fails or becomes inaccessible while the primary is active, the primary
+            segment tracks database changes in WAL logs that are applied to the mirror when it is
+            recovered. For information about segment fault detection and the recovery process, see
+                <xref href="g-detecting-a-failed-segment.xml#topic9"/>. </p>
+        <p>These Greenplum Database system catalog tables contain mirroring and replication
+            information. <ul id="ul_nn3_yjm_rdb">
+                <li>The catalog table <codeph>gp_segment_configuration</codeph> contains the current
+                    configuration and state of primary and mirror segment instances and the master
+                    and standby master instance. </li>
+                <li>The catalog view <codeph>gp_stat_replication</codeph> contains metadata of the
```
Comment (jimmyyih) [inline gpdb-doc/dita/admin_guide/highavail/topics/g-overview-of-segment-mirroring.xml]: > contains metadata

Should probably change to "contains replication statistics".

File: gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml (hunk 9 above)
Comment (jimmyyih) [reply gpdb-doc/dita/ref_guide/system_catalogs/gp_stat_replication.xml]: The 4 valid wal sender states are startup, catchup, streaming, and backup.

File: gpdb-doc/dita/ref_guide/system_catalogs/gp_segment_configuration.xml [hunk 13]
```diff
@@ -102,18 +102,17 @@
             </entry>
             <entry colname="col2">text</entry>
             <entry colname="col3"/>
-            <entry colname="col4">The hostname used to access a particular segment on a segment
-              host. This value may be the same as <codeph>hostname</codeph> in systems upgraded from
-              3.x or on systems that do not have per-interface hostnames configured.</entry>
+            <entry colname="col4">The hostname used to access a particular segment instance on a
+              segment host. This value may be the same as <codeph>hostname</codeph> on systems that
```
[the remark this answers is not in the document]
Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/system_catalogs/gp_segment_configuration.xml]: If a host has multiple NICs, it is the name assigned to the NIC.

**Pull request merged**